### PR TITLE
nitryl and stimulum merge in nitrium

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Re-take of #57384 that i stopped cause i was busy

This PR is the first of a few were i'll be merging similar working gases into one.
This time is Nitryl and Stimulum. They'll be merged into Nitrium , a brown gas with both features of the two gases
The main scope is to add dept to atmos by removing bloated content and/or repeated content that has never seen the lights of the day (how many times have you seen both gases made and used at the same time?)
The PR so far:
-removed nitryl
-removed stimulum
-merged them into Nitrium (Nitrium is now made with trit, nitrogen and bz from a minimum temperature of 1500 K)
-made Nitrium have both gases features such as fast movements and sleep and stun immunity but increased damage taken
-Nitrium can make crystals that spread the chemicals with a cloud (is still far more efficient to just breathe the gas)

## Why It's Good For The Game
Less rare gases, going towards a better atmos gameplay loop

## Changelog

:cl:
del: removed nitryl
del: removed stimulum
balance: merged the two into nitryum that shares the two gases features and that can make a crystal of itself in the crystallizer, the gas is made with trit, nitrogen and bz at over 1500 K
balance: the gas effect can be chosen by the partial pressure of the gas, low pressure and you have the old nitryl effect, higher and you'll have both
/:cl:

### CODE REVIEW

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
I'm kind of curious about your particular reasons for doing this rather then removing one or the other, at least since gas count doesn't impact react cost anymore. You mind elaborating on that?

**Comment by Pickle-Coding** [comment]
If it decomposes into hydrogen, it should release radiation.

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -405,31 +405,31 @@
...
+	id = "nitrium_decomp"
 
-/datum/gas_reaction/nitryl_decomposition/init_reqs()
+/datum/gas_reaction/nitrium_decomposition/init_reqs()
 	requirements = list(
 		/datum/gas/oxygen = MINIMUM_MOLE_COUNT,
-		/datum/gas/nitryl = MINIMUM_MOLE_COUNT,
-		"MAX_TEMP" = 600
+		/datum/gas/nitrium = MINIMUM_MOLE_COUNT,
+		"MAX_TEMP" = 45
```
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: yeah, but max of 45 means normal enviroments can take it

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 2]
```diff
@@ -438,36 +438,34 @@
...
 
-/datum/gas_reaction/nitrylformation/init_reqs()
+/datum/gas_reaction/nitrium_formation/init_reqs()
 	requirements = list(
-		/datum/gas/oxygen = 10,
+		/datum/gas/tritium = 20,
 		/datum/gas/nitrogen = 10,
 		/datum/gas/bz = 5,
-		"MIN_TEMP" = 1500,
-		"MAX_TEMP" = 10000
```
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:449]: I don't like changing things for weakish reasons if we don't know why they were done in the first place

[the remark this answers is not in the document]
Comment (Ghilker) [reply code/modules/atmospherics/gasmixtures/reactions.dm:449]: I added that when doing the 5 new gases lol, just added it for the sake of it https://github.com/tgstation/tgstation/pull/53013#discussion_r479633090 just after this i made the commit and i added a min and max temp, no reason to it, just added it.

[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:449]: lmao

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 3]
```diff
@@ -401,32 +401,33 @@
...
+	id = "nitrium_decomp"
 
-/datum/gas_reaction/nitryl_decomposition/init_reqs()
+/datum/gas_reaction/nitrium_decomposition/init_reqs()
 	requirements = list(
 		/datum/gas/oxygen = MINIMUM_MOLE_COUNT,
-		/datum/gas/nitryl = MINIMUM_MOLE_COUNT,
-		"MAX_TEMP" = T0C + 70 //Pretty warm, explicitly not fire temps. Time bombs are cool, but not that cool. If it makes you feel any better it's close
+		/datum/gas/nitrium = MINIMUM_MOLE_COUNT,
+		"MAX_TEMP" = 45
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:413]: Still not happy with this

Comment (Ghilker) [reply code/modules/atmospherics/gasmixtures/reactions.dm:413]:
```suggestion
		"MAX_TEMP" = T0C + 70 //Pretty warm, explicitly not fire temps. Time bombs are cool, but not that cool. If it makes you feel any better it's close
```

Comment (Ghilker) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: i've set it as nitryl was

**Comment by LemonInTheDark** [comment]
On further thought, why merge the two? If the goal is to simplify things one gas with two sets of effects seems a bit unhelpful.

I realize we're in a bit of a funny position because stim is used as an endgame crafting tool, and nitryl is budget stim and all, but I'm not really sure what this accomplishes outside of just lowering the gas count. 
I'm happy to do that, I don't like the idea of many gases that are just straight upgrades of each other, mostly because atmos isn't the same as say chem, and we can't have big progression trees. But in this case it'd be done to make things easier to understand, and it seems like we're ending up with a frankengas

IDK maybe I'm overthinking this. What do you think?

**Comment by Ghilker** [comment]
From the discord answer
I'm doing it because I'd like to reduce the gas count to have more do to with each gases (increase depth for each one of them) before adding more back. Stim and Nitryl have a similar effect that is to improve movement, so I started with merging them into one gas that has more in it than the two gases separated. I think it would be much better to reduce the gas number by merging them to increase the pool of usability for each one of them and reduce the gas doing the same thing (like H2 and Trit, was thinking of having only H2 but is debatable) before adding more.

> I'm happy to do that, I don't like the idea of many gases that are just straight upgrades of each other, mostly because atmos isn't the same as say chem, and we can't have big progression trees. 

This is the main goal, I don't want similar gases that are weak upgrades/downgrades to others, I just figured to start with these two

**Pull request merged**